bug: Adjust encoder frame rate based on max_fps(/src/main.rs)

lenitain opened the pull request:
Set encoder frame rate based on max_fps argument if provided.

Problem: The video encoder always uses the display refresh rate as its own framerate, while the `--max_fps` argument only reduces the number of frames sent to the encoder. This mismatch results in blurry frames, particularly when the `--max_fps` is significantly lower than the display refresh rate.

Tested on: fedora43 linux + sway

Let me know if anything needs to be adjusted.

lenitain committed:
bug: Adjust encoder frame rate based on max_fps(/src/main.rs)

Set encoder frame rate based on max_fps argument if provided.

russelltg reviewed (changes requested):
I'm somewhat surprised that this affects quality at all, but a good find regardless.

lenitain committed:
Improve encoder framerate selection logic

Better handle fractional framerate.
Select min(max_fps, framerate) as encoder framerate.

russelltg reviewed (changes requested):
sorry i wasn't clearer about exactly how i wanted this done

lenitain reviewed (commented):
Sorry,I did not get you clearly earlier.

lenitain committed:
Update encoder framrate selection

Co-authored-by: Russell Greene <russellgreene8@gmail.com>
